src/main/java/me/leoko/advancedban/bungee/BungeeMethods.java

```diff
@@ -186,13 +186,15 @@ public ProxiedPlayer getPlayer(String name) {
         return ProxyServer.getInstance().getPlayer(name);
     }
 
-    @SuppressWarnings("deprecation")
+
     @Override
     public void kickPlayer(String player, String reason) {
-        if (Universal.get().useRedis()) {
+        if(BungeeMain.getCloudSupport() != null){
+            BungeeMain.getCloudSupport().kick(getPlayer(player).getUniqueId(), reason);
+        }else if (Universal.isRedis()) {
             RedisBungee.getApi().sendChannelMessage("AdvancedBan", "kick " + player + " " + reason);
         } else {
-            getPlayer(player).disconnect(reason);
+            getPlayer(player).disconnect(new TextComponent(reason));
```

Comment:
Are you sure it actually will show the color codes correctly? Shouldn't it be something like `TextComponent.fromLegacyText(...)`?